doc/iterations/2023.10/shapes/contribute_custom_frontend_tool.adoc

```diff
@@ -0,0 +1,42 @@
+= (S) Contribute custom frontend tool
+
+== Problem
+
+Today there is no way to add custom complexe tool to a diagram palette.
+It's only possible to add semantic actions on a clickHandler, but not to add more complex scenarios like open a modal to select data.
+
+== Key Result
+
+Add the possibility to programmatically contribute custom React Components tools to a palette.
+
+== Solution
+
+Add a new React component `DiagramPaletteToolContribution` with two props:
+
+* `canHandle(diagramId: string, diagramElementId: string)`->defined if the custom tool must be added to the palette.
+* `component(props: DiagramPaletteToolContributionComponentProps)`->the custom React component to add to the palette.
+
+With the following available props :
+
+[source]
+----
+editingContextId: string;
+diagramId: string;
+diagramElementId: string;
```

Comment:
Can't we pass the actual `GQLDiagram` and `GQLDiagramElement` (which does not currently exist but should be a common supertype for `GQLNode` and `GQLEdge` which share a lot)?
I'm not sure the ids are enough to implement non-trivial tools.
For example, if I want a tool "Ungroup/unwrap" to be available only on diagram nodes which contain sub-nodes, can I _resolve_ the `diagramElementId` into the actual `GQLNode` to check for that?

Reply:
Done, I added more available data to the canHandle like the kind for a node palette.